substrate/frame/contracts/src/tests.rs

```diff
@@ -3078,7 +3078,7 @@ fn gas_estimation_call_runtime() {
 		// contract encodes the result of the dispatch runtime
 		let outcome = u32::decode(&mut result.result.unwrap().data.as_ref()).unwrap();
 		assert_eq!(outcome, 0);
-		assert!(result.gas_required.ref_time() > result.gas_consumed.ref_time());
+		assert!(result.gas_required.ref_time() >= result.gas_consumed.ref_time());
```

Comment:
This assertion is not incorrect. The comment a bit further up explains this. It is the point of the test to make sure that the pre-charging works. In this case when calling into the runtime. We use a function that has a much bigger max weight than actual weight to exercise that.

Why this broke from your changes is indeed strange.